serde/de.py

```diff
@@ -6,6 +6,7 @@
 import abc
 import dataclasses
 import functools
+import traceback
 import typing
```

Comment:
For Python 3.7 support, Literal has to come from the typing-extensions package.

Could maybe do something like:

```
try:
    from typing import Literal
except ImportError:
    from typing_extensions import Literal
```

Or

https://github.com/python/mypy/issues/8155

```
import sys

if sys.version_info[:2] >= (3, 8):
    from typing import Literal
else:
    from typing_extensions import Literal
```